tests/unit/test_desy.py

```diff
@@ -111,23 +114,29 @@ def test_pipeline(generated_records, expected_records):
         override_generated_fields(generated_record)
         for generated_record in generated_records
     ]
-    sorted_generated_records = deep_sort(clean_generated_records)
-    sorted_expected_records = deep_sort(expected_records)
-    assert sorted_generated_records == sorted_expected_records
-
+    assert DeepDiff(clean_generated_records, expected_records,
+                    ignore_order=True, report_repetition=True) == {}
 
 def test_faulty_marc():
     spider = create_spider()
     path = os.path.abspath('tests/unit/responses/desy/faulty_record.xml')
     with open(path, 'r') as xmlfile:
         data = xmlfile.read()
     result = spider._parsed_items_from_marcxml([data])
-    expected_exception = (
-        "DoJsonError(u'Error in rule \"imprints\" for field \"260__\"',"
-        " (u'Unknown string format:', 'Not a date'), GroupableOrderedDict((("
-        "'a', 'Hamburg'), ('c', 'Not a date'), "
-        "('b', 'Verlag Deutsches Elektronen-Synchrotron'))))"
-    )
+    if sys.version_info[0] < 3:
```

Comment:
can you weaken the test instead? this is testing that exceptions inside the spider due to bad MARC:
1. don't cause the spider to blow up
2. are correctly recorded

So I would do something simple like this instead:
```python
assert result[0].exception.startswith('DoJsonError')
```